database/database.go

```diff
@@ -263,34 +254,20 @@ func (db *DB) managedUsersByField(ctx context.Context, fieldName, fieldValue str
 	return users, nil
 }
 
-// connectionOptionsFromEnv retrieves the DB connection credentials from the
-// environment and returns them in a map.
-func connectionOptionsFromEnv() (map[string]string, error) {
-	opts := make(map[string]string)
-	for _, varName := range []string{envDBHost, envDBPort, envDBUser, envDBPass} {
-		val, ok := os.LookupEnv(varName)
-		if !ok {
-			return nil, errors.New("missing env var " + varName)
-		}
-		opts[varName] = val
-	}
-	return opts, nil
-}
-
 // connectionString is a helper that returns a valid MongoDB connection string
 // based on the passed credentials and a set of constants. The connection string
 // is using the standalone approach because the service is supposed to talk to
 // the replica set only via the local node.
 // See https://docs.mongodb.com/manual/reference/connection-string/
-func connectionString(user, pass, host, port string) string {
+func connectionString(creds DBCredentials) string {
 	// There are some symbols in usernames and passwords that need to be escaped.
 	// See https://docs.mongodb.com/manual/reference/connection-string/#components
 	return fmt.Sprintf(
 		"mongodb://%s:%s@%s:%s/?compressors=%s&readPreference=%s&w=%s&wtimeoutMS=%s",
-		url.QueryEscape(user),
-		url.QueryEscape(pass),
-		host,
-		port,
+		url.QueryEscape(creds.User),
+		url.QueryEscape(creds.Password),
+		creds.Host,
+		creds.Port,
 		mongoCompressors,
```

Comment:
I had to google `mongoCompressors` - never encountered those before - do we need those?

Reply:
We don't strictly need it but since it allows compression of the communication between the replica set nodes it seemed like a good idea. Is there a reason not to have it? I don't have a strong opinion on this one and we can always change it.